src/EnergyPlus/InputProcessing/CsvParser.cc

```diff
@@ -187,40 +162,33 @@ json CsvParser::parse_csv(std::string_view csv, size_t &index, bool &success)
             break;
         } else {
             if (check_first_row) {
+                // Parse the header first, it could have an extra '()' for shading in 22.2.0 and below
+                if (has_header) {
+                    parse_header(csv, index, header);
+                }
                 int num_columns = find_number_columns(csv, index);
```

Comment:
And I reverse the order of parse_header and find_number_of_columns, so that the number of columns I'll be expected is the actual number of records, not the header.

https://github.com/NREL/EnergyPlus/blob/1b333b500c96960b1a1978657b9a6a4a28c9dd72/tst/EnergyPlus/unit/Resources/shading_data_2220.csv#L1-L3

That means when I'm done, headers will be size 3, but columns will only be 2!